plugins/kotlin-serialization/kotlin-serialization-compiler/src/org/jetbrains/kotlinx/serialization/compiler/backend/common/AbstractSerialGenerator.kt

```diff
@@ -61,4 +62,12 @@ abstract class AbstractSerialGenerator(val bindingContext: BindingContext, val c
 
     protected fun ClassDescriptor.getFuncDesc(funcName: String): Sequence<FunctionDescriptor> =
         unsubstitutedMemberScope.getDescriptorsFiltered { it == Name.identifier(funcName) }.asSequence().filterIsInstance<FunctionDescriptor>()
+
+    private fun canUseFieldMissingOptimization(): Boolean {
+        val implementationVersion = VersionReader.getVersionsForCurrentModuleFromContext(
```

Comment:
I think we need to think it over because anyway check optimization won't work in JS IR/native.
M.b. for IR inside `canUseFieldMissingOptimization` check existing of `kotlinx.serialization.internal.PluginExceptionsKt#throwMissingFieldException` and `kotlinx.serialization.internal.PluginExceptionsKt#throwArrayMissingFieldException` functions?

Reply:
Not sure if this possible, but if it is can be easily done, then it sounds like a reasonable way

Reply:
For IR created this behavior: if functions `throwMissingFieldException` and `throwArrayMissingFieldException` are found - then in the internal constructor using golden mask field missing check, otherwise - old checking field by field.
So for IR, golden mask check will be enabled for runtime versions 1.1.0-M1, 1.1.0-RC, 1.1.0, and above.
But external serializers will require a minimum of 1.1.0-M1 version of runtime.